src/SourceFile.js

```diff
@@ -0,0 +1,229 @@
+     * being linted
+     * @param {Function} [options.getLogger] a callback function provided by
+     * the linter to retrieve the log4js logger
+     * @param [String] [options.type] the type of this file
+     * @constructor
+     */
+    constructor(uri, options) {
+        if (!uri) {
+            throw new Error("Attempt to create a SourceFile without a file path");
+        }
+        this.filePath = uri;
+        this.logger = (typeof(options.getLogger) === "function") ? options.getLogger("ilib-lint-common.SourceFile") : undefined;
+        this.sourceLocale = options?.sourceLocale || "en-US";
+        this.type = options?.type || "";
+    }
+
+    /**
+     * Read the contents of the file into memory.
+     * @throws Error if the file could not be read, or if it is not encoded in UTF-8
+     * @protected
+     */
+    read() {
+        // do not convert to a string for the raw property
+        this.raw = fs.readFileSync(this.filePath);
+
+        // ... but do convert for the string property
+        this.content = this.raw.toString("utf-8");
+
+        // mark as not modified with respect to the source
+        this.dirty = false;
+    }
+
+    /** 
+     * A callback function provided by the linter to retrieve the log4js logger
+     * @type {Function | undefined}
+     * @protected
+     */
+    #logger;
+
+    /**
+     * URI or path to the source file. 
+     * @type {String}
+     * @protected
+     */
+    #filePath;
+
+    /**
+     * Get the URI or path to this source file.
+     * 
+     * @returns {String} URI or path to this source file
+     */
+    getPath() {
+        return this.filePath;
+    }
+
+    /**
+     * The raw bytes of the file stored in a Buffer.
+     * @type {Buffer}
+     * @protected
+     */
+    #raw;
+    
+    /**
+     * Return the raw contents of the file as a Buffer of bytes.
+     * This has not been converted into a Unicode string yet.
+     * @returns {Buffer} a buffer containing the bytes of this file
+     */
+    getRaw() {
+        if (this.raw === undefined) {
+            this.read();
+        }
+        return this.raw;
+    }
+
+    /**
+     * The content of the file, stored as regular Javascript string
+     * encoded in UTF-8.
+     *
+     * @type {String}
+     * @protected
+     */
+    #content;
+    
+    /**
+     * Get the content of this file encoded as a regular Javascript
+     * string.
+     * @returns {String} the content of the file, encoded as a JS string
+     */
+    getContent() {
+        if (this.content === undefined) {
+            this.read();
+        }
+        return this.content;
+    }
+    
+    /**
+     * Get the content of this file, encoded as an array of lines. Each
+     * line has its trailing newline character removed.
+     * @returns {Array.<String>} the content as an array of lines
+     */
+    getLines() {
+        if (this.content === undefined) {
+            this.read();
+        }
+        return this.content.split(/\n/g);
```

Comment:
Maybe mention in comment that only LF is supported?

Reply:
Added to the setLines description